### PR TITLE
Lazy loading Google Maps script + maps initialization

### LINKED ISSUE #36: Map load speed
Is there any way to enable lazy loading of the maps used? The map plugin seems to be the cause of speed issues in the chrome audit.
image

### PR DESCRIPTION
**Pull request by Gerfaut**
Using Intersection Observer API. 
Resolve #36 and optimize LightHouse result. 
Let me know what you think :wink:

Thanks a lot for this great lib! 👍 

_Gerfaut_

### CODE REVIEW

File: lib/jekyll-maps/google_map_api.rb [hunk 1]
```diff
@@ -31,7 +48,50 @@ def load_google_maps_api
...
+                } else {
+                    enableMapsObserver();
+                }
+            });
+            
+            function enableMapsObserver() {
+              // Enable Observer on all Maps
+              var maps = document.getElementsByClassName('jekyll-map');
+
+              const observer = new IntersectionObserver(function(entries, observer) {
```
Comment (ayastreb) [inline lib/jekyll-maps/google_map_api.rb:68]: `IntersectionObserver` is not supported in IE, right? Can we either do a feature check and only do lazy loading if `IntersectionObserver` is supported, or include a polyfill?

Comment (Gerfaut) [reply lib/jekyll-maps/google_map_api.rb:68]: Yes, that's correct. 
Sure, I'll have a look to do that and see how I can manage it 😉 

I'll let you know as soon as it is ready!

_Gerfaut_

**Comment by ayastreb** [comment]
Also, I've kicked TravisCI manually and it run the build on this pull request, with a couple of tests failing:
https://travis-ci.org/github/ayastreb/jekyll-maps/jobs/670898756?utm_medium=notification&utm_source=github_status

Could you please check it and make sure tests/lints are passing?

Thanks!

**Comment by Gerfaut** [comment]
Hello @ayastreb, 
I fixed / added tests and did a fallback solution for older browsers not supporting IntersectionObserver (tested on IE11 and working). 
Let me know if there is anything I can help with 😉 

Have a nice day, 

_Gerfaut_

**Comment by ayastreb** [comment]
@Gerfaut sorry for the delay!

Thanks for contribution, published as v2.4.0!

**Comment by ghost** [comment]
Hey @ayastreb, no worries! 

Great news, thanks a lot! 
I hope it will please some features requested ;) 

Let me know if I can help to include some examples in the demo site!

Have a nice day, 

_Gerfaut_

**Comment by ayastreb** [comment]
Hi @Gerfaut ,

the demo [site](https://ayastreb.me/jekyll-maps/) is built from [gh-pages-src](https://github.com/ayastreb/jekyll-maps/tree/gh-pages-src) branch in this repo, and it's a Jekyll site :)

Feel free to submit MR to that branch with changes you'd like to see, although I'm not sure how to show lazy loading on demo site, since it's already has some maps :)

**Comment by ghost** [comment]
Hi @ayastreb, 
Thanks for the info! 
Indeed lazy loading is already "shown" on the demo website 😄 
I'll work on examples for styles (regarding the PR #45) and just add some words about lazy loading / compatibility IE, etc 😉 

I'll submit a PR end of the week probably when it will be ready!

Thanks,

_Gerfaut_

**Pull request merged**